### DISCUSSION TITLE
Cannot connect to DB using Supavisor on Vercel

### DISCUSSION THREAD

**Discussion by ddavisj** [supabase/supabase#20426] [Questions]
Hi, I've spent the last 3 days trying to fix an issue on a new site called growBook and am going a bit crazy trying to work out why it's not working. growBook is based on a Nuxt 3 course from Udemy and makes use of Supabase and Prisma. 

The code is live on GitHub: https://github.com/ddavisj/growBook

When I run growBook locally the site and all database connections work as expected when I use PgBouncer. I'm trying to set up hosting on Vercel's Edge Network using Supabase's Supavisor as PgBouncer support is about to be removed.

If I run the site on Vercel but not on their Edge network, it loads but database calls do not work. My understanding is that I need to run it on the Edge Network in order to be able to make database requests and I'm trying to use Supavisor's pooling connections to do so. 

## Changes for Vercel's Edge Network

As far as I can tell, I've followed all the required recommendations to run the site on the Edge Network. 

a) I've updated the DATABASE_URL env variable:
postgres://postgres.[PROJECT-ID]:[PASSWORD]@aws-0-ap-southeast-2.pooler.supabase.com:6543/postgres

b) I've added the env variable for Vercel edge:
SERVER_PRESET=vercel_edge

c) I've also changed the build command as recommended to:
nuxt build --preset=vercel_edge

d) I've updated the Nuxt config for Nitro (Nuxt's server):
```
   nitro: {
      preset: 'vercel-edge',
      vercel: {
         regions: ['syd1'],
      },
   },
```

## Error: When deployed on Vercel using Supavisor

My site deploys as normal and is live at: https://growbook.vercel.app/

When I load this site using the above settings, I get the following:

> This Edge Function has crashed.
> Your connection is working correctly.
> Vercel is working correctly.
> 500: INTERNAL_SERVER_ERROR
> Code: EDGE_FUNCTION_INVOCATION_FAILED
> ID: syd1::j9pw9-1705276754984-b84f392bbb7c

When I check Vercel's logs, I get the following error:

TypeError: Cannot read properties of undefined (reading 'indexOf')
    at (index.mjs:1:178310)

I checked my site for any use of indexOf and there are none - so I assume this error is related to Nuxt's code.

## Error: When deployed locally using Supavisor

When I load the site locally using Supavisor, I get the following error in my Linux terminal:
- code: "26000", message: "prepared statement \"s0\" does not exist"

---

Thanks for reading. Any suggestions or advice would be much appreciated!

---

EDIT: I created a staging site that uses Vercel, not Vercel Edge and it's live but quite slow here: https://growbook-stage.vercel.app/

**Comment by ddavisj** [comment]
Ok. Managed to sort of get this working with Vercel but without Edge.

1) I just removed the _edge and -edge from all commands above. If anyone can help me upgrade to Edge, it'd make things a lot faster and better (as it stands, the app is almost too slow to run without Edge). See https://nitro.unjs.io/deploy/providers/vercel

2) When building with Prisma and working with Vercel, you need to add a "prisma generate" command to the build step. Think it's because Vercel caches the build, and this means that Prisma won't be generated unless specified in the build step. See https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/vercel-caching-issue

**Comment by GaryAustin1** [comment]
You should skim/search this thread and see if anyone else mentions that. https://github.com/orgs/supabase/discussions/17817

**Reply by ddavisj** [reply]
Thanks! I had a look a little while back but seems there's a lot more posts.. May be some gold in there

**Comment by LyghtCode** [comment]
I am having a similar issue where i can only connect locally but on my Vercel deployment the database doesn't connect
